lib/pure/logging.nim

```diff
@@ -320,7 +320,7 @@ proc substituteLog*(frmt: string, level: Level, args: varargs[string, `$`]): str
 
 method log*(logger: Logger, level: Level, args: varargs[string, `$`]) {.
             raises: [Exception], gcsafe,
-            tags: [TimeEffect, WriteIOEffect, ReadIOEffect], base.} =
+            tags: [TimeEffect, WriteIOEffect, ReadIOEffect, RootEffect], base.} =
```

Comment:
Good job, however, since ``RootEffect`` is the base for everything, ``tags: [RootEffect]`` suffices.

Reply:
Ah okay, that makes sense and I should have realized that. I will update the PR when I get home this evening.

Reply:
Fixed per your suggestion.